### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by rohanmeh**
*Issue #, if available:*

*Description of changes:*
Add intro links, gif, move testing guidance "theory" to separate file, add async test sample list.

Note: Typescript and Java links need to be updated after the PR's for those samples are merged into main.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dancfox — CHANGES_REQUESTED** [review]

**Comment by dancfox** [comment]
I'm conflicted about the "moving test theory into a different file" but am willing to merge it for now and see how it feels.

File: README.md [hunk 1]
```diff
@@ -1,21 +1,46 @@
-## Introduction
-This repository contains sample code demonstrating a variety of test patterns for serverless applications.
+<div align="center">
```
[the remark this answers is not in the document]
Comment (rohanmeh) [reply README.md:1]: The example "good" Readme's use center alignment but it's not a must have for me - would be fine left aligned as well.

There's no way to do center-aligned markdown text without the html (from what i see on stackoverflow)

I agree we can be consistent about not using HTML unless we absolutely have to.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,46 @@
...
+</h2>
+This repo is designed to provide developers with guidance for implementing comprehensive test suites for their serverless applications.
+</br>
+<b>Runtime Guides:</b>
 
-## Testing in the Cloud
-Well-designed serverless applications may employ a variety of testing techniques to satisfy a range of requirements and conditions. However, based on current tooling, we recommend customers **focus on testing in the cloud** as much as possible. While testing in the cloud may create additional developer latency, increase costs, and require some customers to invest in additional dev-ops controls, this technique provides the most reliable, accurate, and complete test coverage. Performing tests in the context of the cloud allows you to test IAM policies, service configurations, quotas, and the most up to date API signatures and return values. Tests run in the cloud are most likely to produce consistent results as your code is promoted from environment to environment.
+<a href="https://github.com/aws-samples/serverless-test-samples/tree/main/python-test-samples">Python</a>
+·
+<a href="https://github.com/aws-samples/serverless-test-samples/tree/main/python-test-samples">Typescript</a>
```
[the remark this answers is not in the document]
Comment (rohanmeh) [reply README.md:11]: I think Java should be ready in the next few days so I can update the link in the PR and then we can merge it in?

Will update the PR with Typescript link.

[the remark this answers is not in the document]
Comment (rohanmeh) [reply README.md:11]: I think we can "launch" the repo with all 3 sample folders (Python Typescript Java) and the updated readme all together.

**Comment by rohanmeh** [comment]
> 

About test theory in separate file, what I saw in other readme's was they didn't have large blocks of texts like we did - they got straight to the point. I think that content will mostly live in ServerlessLand (and other docs pages) in the future, but willing to put it back in if we need to.

**Review by dancfox — APPROVED** [review]

**Pull request merged**